### ISSUE TITLE
Relation to python-duckdb?

### ISSUE THREAD

**Issue by wisp3rwind** [conda-forge/duckdb-feedstock#6]
### Comment:

Hi!

Could you elaborate what the relation of this package is to [python-duckdb](https://github.com/conda-forge/python-duckdb-feedstock)? It seems that they provide the same (namely, the python package, not the standalone cli or libduckdb), but python-duckdb has been around much longer and includes some additional files. Unfortunately, this situation is quite confusing, completely undocumented, and the two packages conflict (but conda doesn't complain about that when installing both).

Thanks!

**Comment by gforsyth** [comment]
@conda-forge/help-python 

this feedstock is a duplicate of https://github.com/conda-forge/python-duckdb-feedstock

**Comment by BastianZim** [comment]
This is probably better with @conda-forge/core

**Comment by ocefpaf** [comment]
B/c both variants are out the best we can do is:

- archive the newer one
- ask if the maintainer(s) of the newer one wants to be added in the old one
- add a multiple outputs to reserve both names so folks using one or the other will always get the latest version (see https://github.com/conda-forge/importlib_metadata-feedstock/blob/main/recipe/meta.yaml)

**Comment by gforsyth** [comment]
@ericdill I'm unclear on why we would archive the older one, since we've got that one all tuned up (and building for 5 platforms instead of just linux-64)

**Comment by ocefpaf** [comment]
> @ericdill I'm unclear on why we would archive the older one, since we've got that one all tuned up (and building for 5 platforms instead of just linux-64)

Same. My comment was specific to keep the status quo of the first group that submitted it. In fact, the clobbering new package was a mistake from our end and it should've never been merged to start with.

PS: @ericdill  you  mind adding a new comment instead of editing mine?

**Comment by ericdill** [comment]
Yeah fair enough my bad. Figured it was a typo

**Comment by ocefpaf** [comment]
> Yeah fair enough my bad. Figured it was a typo

My grammar/spelling may be quite bad but I hope the overall meaning of archiving the new one and porting names and maintainers to the older one is capture in my comment. I reverted the edit above and hope that makes sense @conda-forge/duckdb and @conda-forge/python-duckdb teams.

**Comment by ericdill** [comment]
Hey now that im back at a computer, want to give you a more thorough response. First and foremost, sorry. I had the wrong mental model (been spending way too much time editing in google docs) when reading this thread and was on autopilot. My apologies for adding confusion to this thread, @gforsyth and @ocefpaf, and possibly personally offending you, @ocefpaf.

**Comment by ocefpaf** [comment]
@ericdill, all good. Editing for small fixes is fine. It was just confusing in this case b/c you reversed the logic and we did not know the context for that.

**Comment by sugatoray** [comment]
So, what's the verdict on this one?
It makes sense to maintain only one package, instead of two.

cc: @ocefpaf @ericdill @BastianZim

**Comment by ocefpaf** [comment]
> archiving the new one and porting names and maintainers to the older one is capture in my comment.

If you all agree we should start implementing this solution.

**Comment by dhirschfeld** [comment]
In case it helps, here's an example recipe publishing a package under 2 different names:
https://github.com/conda-forge/arrow_odbc-feedstock/blob/042fa11dec4119cfc3782c21c4b9180504da4724/recipe/meta.yaml#L48-L62

**Comment by ocefpaf** [comment]
> In case it helps, here's an example recipe publishing a package under 2 different names: https://github.com/conda-forge/arrow_odbc-feedstock/blob/042fa11dec4119cfc3782c21c4b9180504da4724/recipe/meta.yaml#L48-L62

That is probably the cleanest example we have... We should put that one in the docs :grimacing: 
(Shameless ping to @jaimergp who was looking into docs a while back.)

**Comment by jaimergp** [comment]
TIL you can do aliased packages like that. I'll start a PR documenting this, thanks!

**Comment by isuruf** [comment]
> That is probably the cleanest example we have.

Not really. It's a bad example. Because of the exact=True pin there, it's only installable in linux-64.

**Comment by ocefpaf** [comment]
> Not really. It's a bad example. Because of the exact=True pin there, it's only installable in linux-64.

Not sure I follow. If that b/c of hashes in the version? Would pinning `x.x.x` be better?

PS: when I mentioned it was a good example it was in comparison to what I did in the past like https://github.com/conda-forge/importlib_metadata-feedstock/pull/37

**Comment by isuruf** [comment]
> Not sure I follow. If that b/c of hashes in the version?

`arrow-odbc` with the dash is a noarch: python package with the same hash for different OSes. Therefore only the package that is built the fastest gets uploaded. i.e. linux-64. However the package depends on the exact hash of `arrow_odbc` with the underscore which depends on the OS. Therefore `arrow-odbc` with the dash is *not* installable on `osx-64` even though `arrow_odbc` with the underscore is installable.

> Would pinning x.x.x be better?

Yes, if you don't anticipate changes based on build number. `arrow_odbc {{ version }} *_{{ build_num }}` is best.

**Comment by ocefpaf** [comment]
Got it! thanks for the explanation. (And more stuff we should document!)

**Comment by jaimergp** [comment]
Started PR at https://github.com/conda-forge/conda-forge.github.io/pull/1989. Comments welcome!

**Comment by dhirschfeld** [comment]
> Yes, if you don't anticipate changes to build number. `arrow_odbc {{ version }} *_{{ build_num }}` is best.

@isuruf - is this related to https://github.com/conda/conda-build/issues/4415

**Comment by isuruf** [comment]
Nope.

**Comment by dhirschfeld** [comment]
NB: Over in [`arrow_odbc`](https://github.com/conda-forge/arrow_odbc-feedstock/pull/23) it seemed that the required syntax is `arrow_odbc ={{ version }}=*_{{ build_num }}`

**Comment by isuruf** [comment]
What is the error you get with that syntax?

**Comment by dhirschfeld** [comment]
I added a test commit - [conda-forge/arrow_odbc-feedstock@`4af5941` (#23)](https://github.com/conda-forge/arrow_odbc-feedstock/pull/23/commits/4af5941469eb2a64e96a00652ce9e01e993d2c08)

The error is:
```
conda_build.exceptions.RecipeError: Mismatching hashes in recipe. Exact pins in dependencies that contribute to the hash often cause this. Can you change one or more exact pins to version bound constraints?
Involved packages were:
Mismatching package: arrow_odbc (id py310hcb5633a_1); dep: arrow_odbc 1.2.0 *_1; consumer package: arrow-odbc
```

**Comment by jonashaag** [comment]
Hi all, I was confused about this today and even began to fix the recipe in this repo for the more recent releases until I noticed that this feedstock is deprecated.

Can we mark this feedstock deprecated somehow?

**Comment by BastianZim** [comment]
If the maintainers agree, we can archive it: https://conda-forge.org/docs/maintainer/updating_pkgs.html#archiving-feedstocks

CC @sugatoray

**Comment by sugatoray** [comment]
@BastianZim Yes. Fine by me. Please add me to the list of maintainers to the other feedstock, before this feedstock is archived.

**Comment by BastianZim** [comment]
I'm just a passerby 😄 Anything else needs to be done by the respective feedstock owners.

**Comment by gforsyth** [comment]
@sugatoray -- if you put in a PR adding yourself to the maintainer list of `python-duckdb` I can merge it in.

Also feel free to take a crack at getting the 0.9.0 release passing CI 😁

**Comment by jonashaag** [comment]
Done. Can we archive this now please?

**Comment by xhochy** [comment]
Yes, please make a PR to admin-requests.
